server/server.js

```diff
@@ -1,15 +1,110 @@
+const cors = require('cors')
+
+app.use(cors())
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
 
 // Store and retrieve your videos from here
 // If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+let videos = [
+  {
+    id: 523523,
+    title: "Never Gonna Give You Up",
+    url: "https://www.youtube-nocookie.com/embed/dQw4w9WgXcQ",
+    rating: 23,
+  },
+  {
+    id: 523427,
+    title: "The Coding Train",
+    url: "https://www.youtube-nocookie.com/embed/HerCR8bw_GE",
+    rating: 230,
+  },
+  {
+    id: 82653,
+    title: "Mac & Cheese | Basics with Babish",
+    url: "https://www.youtube-nocookie.com/embed/FUeyrEN14Rk",
+    rating: 2111,
+  },
+  {
+    id: 858566,
+    title: "Videos for Cats to Watch - 8 Hour Bird Bonanza",
+    url: "https://www.youtube-nocookie.com/embed/xbs7FT7dXYc",
+    rating: 11,
+  },
+  {
+    id: 453538,
+    title:
+      "The Complete London 2012 Opening Ceremony | London 2012 Olympic Games",
+    url: "https://www.youtube-nocookie.com/embed/4As0e4de-rI",
+    rating: 3211,
+  },
+  {
+    id: 283634,
+    title: "Learn Unity - Beginner's Game Development Course",
+    url: "https://www.youtube-nocookie.com/embed/gB1F9G0JXOo",
+    rating: 211,
+  },
+  {
+    id: 562824,
+    title: "Cracking Enigma in 2021 - Computerphile",
+    url: "https://www.youtube-nocookie.com/embed/RzWB5jL5RX0",
+    rating: 111,
+  },
+  {
+    id: 442452,
+    title: "Coding Adventure: Chess AI",
+    url: "https://www.youtube-nocookie.com/embed/U4ogK0MIzqk",
+    rating: 671,
+  },
+  {
+    id: 536363,
+    title: "Coding Adventure: Ant and Slime Simulations",
+    url: "https://www.youtube-nocookie.com/embed/X-iSQQgOd1A",
+    rating: 76,
+  },
+  {
+    id: 323445,
+    title: "Why the Tour de France is so brutal",
+    url: "https://www.youtube-nocookie.com/embed/ZacOS8NBK6U",
+    rating: 73,
+  },
+];
 
-// GET "/"
 app.get("/", (req, res) => {
   // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+  res.send(videos);
 });
+
+app.post("/", (req, res) => {
+  let data = req.body;
+  res.status(200).send(...videos, JSON.stringify(data));
+});
+
+app.get("/:id", (req, res) => {
+  const videoId = +req.params.id;
+  const filteredVideos = videos.filter((video) => video.id === videoId);
+  res.status(200).send(filteredVideos);
+});
+
+app.delete("/delete/:id", (req, res) => {
+  const videoId = +req.params.id;
+  const videoIndex = videos.findIndex((video) => +video.id === videoId);
+  // if the parameter is not exist in the array it shows -1, then i used the following condition
+  if (videoIndex !== -1) {
+    videos.splice(videoIndex, 1);
+    res.status(200).send(videos);
+  } else {
+    res.status(500).send({
+      result: "failure",
+      message: "Video could not be deleted",
+    });
+  }
```

Comment:
This looks like it would work, if the button was wired up to it, but you haven't updated the list in any of the other functions, so added videos will disappear as soon as the user refreshes. Also, unrelated to this, but the video likes dislikes seem to count in twos?